nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -189,7 +189,92 @@ in
       };
 
       buildMachines = mkOption {
-        type = types.listOf types.attrs;
+        type = types.loaOf (submodule ({ config }: {
+          options = {
+            sshUser = mkOption {
+              type = types.nullOr types.string;
+              default = null;
+              description = ''
+                The user as who to SSH to the build machine.
+              '';
+            };
+            hostName = mkOption {
+              type = types.string;
+              description = ''
+                The hostname of the build machine.
+              '';
+            };
+            system = mkOption {
+              type = types.string;
+              default = null;
+              description = ''
+                The system type the build machine can execute derivations on.
+                null if multiple are supported.
+              '';
+            };
+            systems = mkOption {
+              type = types.listOf types.string;
+              default = [];
+              description = ''
+                The system types the build machine can execute derivations on.
+              '';
+            };
+            sshKey = mkOption {
+              type = types.string;
+              default = "-";
+              description = ''
+                The path to the SSH private key with which to authenticate with
+                the build machine. <literal>"-"</literal> indicates falling back
```

Comment:
Probably should either drop the `"` or at least move the quote out of the literal element.